src/main/java/de/tum/in/www1/artemis/service/connectors/hades/dto/HadesBuildResultNotificationDTO.java

```diff
@@ -0,0 +1,121 @@
+package de.tum.in.www1.artemis.service.connectors.hades.dto;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import org.apache.commons.lang3.ObjectUtils;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import de.tum.in.www1.artemis.domain.BuildLogEntry;
+import de.tum.in.www1.artemis.service.connectors.ci.notification.dto.TestwiseCoverageReportDTO;
+import de.tum.in.www1.artemis.service.dto.AbstractBuildResultNotificationDTO;
+import de.tum.in.www1.artemis.service.dto.StaticCodeAnalysisReportDTO;
+
+/**
+ * Note: due to limitations with inheritance, we cannot declare this as a record,
+ * but we can use it in a similar way with final fields.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class HadesBuildResultNotificationDTO extends AbstractBuildResultNotificationDTO {
```

Comment:
You could convert this to a record by using composition (or duplication if needed) instead of inheritance. Stephan already said that that would be recommended.

Reply:
Can I do that without changing AbstractBuildResultNotificationDTO? This is used at multiple other places.

Reply:
Probably not. Since the existing solutions do it this way as well it should be fine.